### ISSUE TITLE
Scheduled job runs multiple times

### ISSUE THREAD

**Issue by DevanshB10** [agenda/agenda#1575]
```
{
  "_id": {
    "$oid": "65d21a62b0672011458b40f9"
  },
  "name": "a52ea4aa10e4ecc8bfae408d90a74d36665ce7e661d35efe380e2828d43177b9",
  "data": null,
  "priority": 0,
  "shouldSaveResult": false,
  "type": "normal",
  "nextRunAt": null,
  "lastModifiedBy": null,
  "lockedAt": {
    "$date": "2024-02-18T15:15:57.428Z"
  },
  "lastRunAt": {
    "$date": "2024-02-18T15:15:57.546Z"
  }
}
```

I scheduled this job for a specific time to run once. But this is executing every 10 minutes, even after achieving nextRunAt: null

Can anyone help, why does this happen?

**Comment by rkrasto** [comment]
How much time does your job take to finish?
I noticed that the 'lockedAt' field is not null, indicating that the job is still running. 
By default, the 'lockLifetime' is set to 10 minutes. Therefore, if the job is still executing for more than 10 minutes, it will restart.

**Comment by DevanshB10** [comment]
@rkrasto the job takes less than 1 second to complete.
For the time being, I am managing this by deleting the job after it runs once. Is there a better way though?

Another question: `agenda.maxConcurrency(50);`

What would this translate to? Does this mean atmost 50 jobs can run at the same time?

**Comment by SantoshKumarRavi** [comment]
@rkrasto yes. i was facing the same issues. it will restart after 10 minutes (default agenda lock time ) , By setting lockedAt null, it will not reschedule the same job.
